File: app/src/main/java/com/boost/espressotest/data/api/mapper/ProductApiToProductContentMapper.java
package com.boost.espressotest.data.api.mapper;


import com.boost.espressotest.data.api.model.ProductApi;
import com.boost.espressotest.data.local.model.ProductEntity;

import io.reactivex.functions.Function;

public class ProductApiToProductContentMapper implements Function<ProductApi, ProductEntity> {

    @Override
    public ProductEntity apply(ProductApi productApi) {
        return new ProductEntity(
                productApi.getId(),
                productApi.getName(),
                productApi.getPriceInCents(),
                productApi.getProducerName(),
                productApi.getImageUrl());
    }
}

File: app/src/main/java/com/boost/espressotest/data/api/mapper/ProductApiToProductContentMapper.kt
package com.boost.espressotest.data.api.mapper

import com.boost.espressotest.data.api.model.ProductApi
import com.boost.espressotest.data.local.model.ProductEntity
import io.reactivex.functions.Function

class ProductApiToProductContentMapper : Function<ProductApi, ProductEntity> {

    override fun apply(productApi: ProductApi): ProductEntity {
        with(productApi) {
            return ProductEntity(
                    id ?: 0,
                    name ?: "",
                    priceInCents ?: 0,
                    producerName ?: "",
                    imageUrl ?: "")
        }
    }
}
